hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/tools/federation/RouterAdmin.java

```diff
@@ -1364,7 +1364,12 @@ public int genericRefresh(String[] argv, int i) throws IOException {
    * @throws IOException if the operation was not successful.
    */
   private int refreshCallQueue() throws IOException {
+    // for security authorization
+    // server principal for this call
+    // should be Routers's one.
     Configuration conf = getConf();
+    conf.set(CommonConfigurationKeys.HADOOP_SECURITY_SERVICE_USER_NAME_KEY,
```

Comment:
Why i did not found HADOOP_SECURITY_SERVICE_USER_NAME_KEY in CommonConfigurationKeys?

Reply:
It is set when the refreshcallqueue is executed. For reference in DFSAdmin we handle it this way https://github.com/apache/hadoop/blob/b5f88990b721aebe84d4bbf97ed2fdfbd0f42854/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/tools/DFSAdmin.java#L1875

Also the key is defined in https://github.com/apache/hadoop/blob/b5f88990b721aebe84d4bbf97ed2fdfbd0f42854/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/CommonConfigurationKeysPublic.java#L668